pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -305,8 +305,41 @@ static __always_inline void process_headers_frame(struct __sk_buff *skb, http2_s
     }
 }
 
-static __always_inline bool http2_entrypoint(struct __sk_buff *skb, skb_info_t *skb_info, conn_tuple_t *tup, http2_ctx_t *http2_ctx) {
-    // Checking for preface
+static __always_inline void parse_frame(struct __sk_buff *skb, skb_info_t *skb_info, conn_tuple_t *tup, http2_ctx_t *http2_ctx, struct http2_frame *current_frame) {
+    http2_ctx->http2_stream_key.stream_id = current_frame->stream_id;
+    http2_stream_t *current_stream = http2_fetch_stream(&http2_ctx->http2_stream_key);
+    if (current_stream == NULL) {
+        skb_info->data_off += current_frame->length;
+        return;
+    }
+
+    if (current_frame->type == kHeadersFrame) { // This is a header frame.
+        process_headers_frame(skb, current_stream, skb_info, tup, &http2_ctx->dynamic_index, current_frame);
+    } else {
+        skb_info->data_off += current_frame->length;
+    }
+
+    if ((current_frame->flags & HTTP2_END_OF_STREAM) == HTTP2_END_OF_STREAM) { // This is end of stream.
```

Comment:
Same for this doc, what is the added value of it? even the variable names say the same

Reply:
removed